addons/account/models/account_move.py

```diff
@@ -858,13 +860,15 @@ def _reconcile_lines(self, debit_moves, credit_moves, field):
                 'currency_id': currency,
             })
 
-        part_rec = self.env['account.partial.reconcile']
         with self.env.norecompute():
             for partial_rec_dict in to_create:
                 new_rec = self.env['account.partial.reconcile'].create(partial_rec_dict)
-                part_rec += new_rec
-                if cash_basis:
-                    new_rec.create_tax_cash_basis_entry(cash_basis_percentage_before_rec)
+                if not cash_basis:
+                    continue
```

Comment:
this IF can be out of the FOR to check it only once. cc @hbto

Reply:
@nhomar I Know I didn't want to change the original estructure of the code.

Reply:
@hbto 
What about merge the if sentences?

Reply:
Besides the IF cannot be out of the FOR loop because the 865 Line creates a partial reconciliation (APR) be it cash_basis True or False, having the IF out of the For loop will avoid the APR record creation.

Reply:
Changes have been done